rust/core-lib/src/styles.rs

```diff
@@ -257,21 +280,19 @@ impl ThemeStyleMap {
         Some(theme_name.to_string())
     }
 
-    /// Load all themes inside the given directory.
+    /// Cache all themes names and their paths inside the given directory.
     pub(crate) fn load_theme_dir(&mut self) {
         if let Some(themes_dir) = self.themes_dir.clone() {
             match ThemeSet::discover_theme_paths(themes_dir) {
                 Ok(themes) => {
                     self.caching_enabled = self.caching_enabled && self.init_cache_dir();
-
+                    // We look through the theme folder here and cache their names/paths to a
+                    // path hashmap.
                     for theme_p in themes.iter() {
-                        match self.try_load_from_dump(theme_p) {
-                            Some((k, v)) => {
-                                self.insert_to_map(k, v, theme_p);
-                            }
-                            None => {
-                                let _ = self.load_theme(theme_p);
-                            }
+                        if let Some(theme_name) = theme_p.file_stem().and_then(OsStr::to_str) {
+                            self.path_map.insert(theme_name.to_string(), theme_p.to_path_buf());
+                        } else {
+                            error!("Invalid theme name, skipping");
```

Comment:
maybe we should also log `theme_p` here, so we know which theme was the source of the error?